### ISSUE TITLE
[FIX-024] Correct misleading metric and parameter-group examples

### ISSUE THREAD

**Issue by thekaveh** [thekaveh/NNx#259]
**Category:** Documentation contracts · **Priority:** P3 · **Size:** S

**Evidence status:** Verified documentation corrections.

**Context and evidence.** [Trainer builder docs](https://github.com/thekaveh/NNx/blob/0c707eb5b80d2173843e60102ba326e4f997749d/src/nnx/trainer/params_builder.py#L96) reverse custom metric arguments and imply validation-only invocation. The implementation uses `(y_true, y_pred)` and the configured training path also computes metrics. [EarlyStopping docs](https://github.com/thekaveh/NNx/blob/0c707eb5b80d2173843e60102ba326e4f997749d/src/nnx/nn/callbacks.py#L83) mention maximizing accuracy/F1 despite a whitelist limited to loss/error. [Parameter-group example](https://github.com/thekaveh/NNx/blob/0c707eb5b80d2173843e60102ba326e4f997749d/src/nnx/finetune/param_groups.py#L41) promises no weight decay on every bias, but first-match selection assigns `encoder.bias` to `encoder.*` and retains `5e-4` decay. These are documentation fixes; arbitrary metric monitors and composable group policies belong to feature design.

**Requested change.** Correct all three examples against current behavior. For the group example, show an explicit `encoder.*bias` rule carrying both reduced LR and zero decay before broader `encoder.*` and remaining `*.bias` rules. Audit mirrored documentation for these exact claims and update only affected surfaces.

**Acceptance criteria.**

- [ ] Custom metric docs consistently use `(y_true, y_pred)` and accurately identify invocation points.
- [ ] EarlyStopping examples use accepted monitor keys and explain `mode` without implying unavailable metric selection.
- [ ] A runnable tiny named-module example demonstrates the promised LR/decay for encoder weight, encoder bias, head weight, and head bias.
- [ ] An asymmetric metric example would fail if its argument order were swapped; samples are checked against the supported API.
- [ ] Public docs/docstrings contain no contradictory copies of the corrected claims; runtime behavior remains backward compatible.

**Sample corrected group rules (existing API).**

```python
specs = [
    NNParamGroupSpec(name_pattern="encoder.*bias", lr_multiplier=.01, weight_decay=0),
    NNParamGroupSpec(name_pattern="encoder.*", lr_multiplier=.01),
    NNParamGroupSpec(name_pattern="*.bias", weight_decay=0),
]
```

**Scope and integration notes.** None. A future generic monitor feature may later expand these examples; do not fix documentation by silently changing selection precedence or metric call signatures.

## Delivery order and dependencies

- **Execution order:** 60 / 67 (priority-aware topological order).
- **Dependency wave:** 1. Items in the same wave may proceed independently when their prerequisites are satisfied.
- **Blocked by:** None within this backlog.
- **Related / optional integration:** [FEAT-003](https://github.com/thekaveh/NNx/issues/213). These links do not block this ticket.

**Dependency rationale.** Independent repair of current behavior. No dependency; FEAT-003 adds monitor capabilities separately.

**Related existing issues (distinct scope):** [#189](https://github.com/thekaveh/NNx/issues/189).

Reviewed against [`0c707eb`](https://github.com/thekaveh/NNx/commit/0c707eb5b80d2173843e60102ba326e4f997749d) on 2026-09-20. Acceptance criteria are prospective; this issue does not claim an implementation or new test pass.
